packaging/datadog-agent/source/setup_agent.sh

```diff
@@ -1,465 +1,529 @@
 #!/bin/sh
+# Bail on errors
+set -e
+# We shouldn't have unbounded vars
+set -u
+
+#######################################################################
+# SCRIPT KNOBS
+#######################################################################
+# Update for new releases, will pull this tag in the repo
+AGENT_VERSION="5.2.2"
+# Pin pip version, in the past there was some buggy releases and get-pip.py
+# always pulls the latest version
+PIP_VERSION="6.0.8"
+VIRTUALENV_VERSION="1.11.6"
+SUPERVISOR_VERSION="3.0b2"
+
+#######################################################################
+# OVERRIDABLE VARIABLES:
+#
+# $DD_API_KEY
+#   Sets your API key in the config when installing.
+#   If not specified, the script will not install a default config.
+#   You can find a sample at $DD_HOME/datadog.conf.example and create
+#   one yourself at $DD_HOME/datadog.conf
+# $DD_HOME
+#   Sets the agent installation directory.
+#   Defaults to $HOME/.datadog-agent
+# $DD_START_AGENT
+#   0/1 value. 1 will start the agent at the end of the script
+#   Defaults to 1.
+# $DD_DOG
+#   0/1 value. 1 will print a cute pup at the beginning of the script
+#   Defaults to 0.
+#
+#
+# $IS_OPENSHIFT DEPRECATED!
+#   Used to be a different setup for OpenShift, it's no a noop, upgrade
+#   your cartridge if you're using it.
+#######################################################################
+set +u # accept temporarily unbound vars, because we set defaults
+# If DD_HOME is unset
+if [ -z "$DD_HOME" ]; then
+    # Compatibilty: dd_home used in lieu of DD_HOME
+    if [ -n "$dd_home" ]; then
+        DD_HOME="$dd_home"
+    else
+        if [ "$(uname)" = "SunOS" ]; then
+            DD_HOME="/opt/local/datadog"
+        else
+            DD_HOME=$HOME/.datadog-agent
+        fi
+    fi
+fi
 
-# figure out where to pull from
-tag="5.2.2"
-
-PIP_VERSION="6.0.6"
+DD_API_KEY=${DD_API_KEY:-no_key}
 
-#######################
-# Define some helpers #
-#######################
+DD_START_AGENT=${DD_START_AGENT:-1}
 
-dogweb_reporting_failure_url="https://app.datadoghq.com/agent_stats/report_failure"
-email_reporting_failure="support@datadoghq.com"
-agent_help_page="http://docs.datadoghq.com/guides/basic_agent_usage/"
-see_agent_on_datadog_page="https://app.datadoghq.com/infrastructure"
+if [ -n "$IS_OPENSHIFT" ]; then
+    printf "IS_OPENSHIFT is deprecated and won't do anything\n"
```

Comment:
Should we stop the execution here and give better instructions or it doesn't matter and will just do the right thing ?